casepro/backend/junebug.py

```diff
@@ -307,5 +336,40 @@ def get_url_patterns(self):
 
         :return: a list of URL patterns.
         """
-        # TODO: Implement views for receiving messages.
-        return []
+        return [
+            url(
+                settings.JUNEBUG_INBOUND_URL, received_junebug_message,
+                name='inbound_junebug_message'
+            ),
+        ]
+
+
+@csrf_exempt
+def received_junebug_message(request):
+    '''Handles MO messages from Junebug.'''
+    if request.method != 'POST':
+        return JsonResponse({'reason': 'Method not allowed.'}, status=405)
+
+    try:
+        data = json.loads(request.body)
+    except ValueError as e:
+        return JsonResponse(
+            {'reason': 'JSON decode error', 'details': e.message}, status=400)
+
+    identity_store = IdentityStore(
+        settings.IDENTITY_API_ROOT, settings.IDENTITY_AUTH_TOKEN,
+        settings.IDENTITY_ADDRESS_TYPE)
+    identities = identity_store.get_identities_for_address(data.get('from'))
+    try:
+        identity = identities.next()
+    except StopIteration:
+        identity = identity_store.create_identity(data.get('from'))
+    contact = Contact.get_or_create(request.org, identity.get('id'))
+
+    message_id = uuid_to_int(data.get('message_id'))
```

Comment:
This function is specified in the `utils` module. It's partially reversible, and I think the only reason we would need to reverse this is for debugging (so maybe partial is okay?).

The issue is that Junebug uses UUIDs for message IDs, while casepro uses an `IntegerField`, which doesn't have enough bits to store an entire UUID. I've done (what I think) is a best case scenario, in order to avoid collisions, and maybe to reverse the lower bits of the UUID, which should be useful for debugging. Suggestions are welcome to improve this though.